test/Target/ExportSMTLIB/attributes.mlir

```diff
@@ -50,6 +51,105 @@ smt.solver () : () -> () {
   smt.assert %3
 
 
+  // CHECK: (assert (let (([[V21:.+]] (exists (([[V22:.+]] Int) ([[V23:.+]] Int))
+  // CHECK:                           ( ! (let (([[V24:.+]] (= [[V22]] [[V23]])))
+  // CHECK:                           (let (([[V25:.+]] (=> [[V24]] true)))
+  // CHECK:                           [[V25]])) 
+  // CHECK:                           :pattern ((let (([[V26:.+]] (= [[V22]] [[V23]])))
+  // CHECK:                           [[V26]]))))))
+  // CHECK:               [[V21]]))
+
+  %6 = smt.exists {
+    ^bb0(%arg2: !smt.int, %arg3: !smt.int):
+    %4 = smt.eq %arg2, %arg3 : !smt.int
+    %5 = smt.implies %4, %true
+    smt.yield %5 : !smt.bool
+  } patterns {
+    ^bb0(%arg2: !smt.int, %arg3: !smt.int):
+    %4 = smt.eq %arg2, %arg3 : !smt.int
+    smt.yield %4: !smt.bool
+  }
+  smt.assert %6
+
+  // CHECK: (assert (let (([[V27:.+]] (exists (([[V28:.+]] Int) ([[V29:.+]] Int))
+  // CHECK:                           ( ! (let (([[V30:.+]] (= [[V28]] [[V29]])))
+  // CHECK:                           (let (([[V31:.+]] (=> [[V30]] true)))
+  // CHECK:                           [[V31]])) :weight 2
+  // CHECK:                           :pattern ((let (([[V32:.+]] (= [[V28]] [[V29]])))
+  // CHECK:                   [[V32]]))))))
+  // CHECK:         [[V27]]))
+
+  %7 = smt.exists weight 2 {
+    ^bb0(%arg2: !smt.int, %arg3: !smt.int):
+    %4 = smt.eq %arg2, %arg3 : !smt.int
+    %5 = smt.implies %4, %true
+    smt.yield %5 : !smt.bool
+  } patterns {
+    ^bb0(%arg2: !smt.int, %arg3: !smt.int):
+    %4 = smt.eq %arg2, %arg3 : !smt.int
+    smt.yield %4: !smt.bool
+  }
+  smt.assert %7
+
+  // CHECK: (assert (let (([[V33:.+]] (exists (([[V34:.+]] Int) ([[V35:.+]] Int))
+  // CHECK:                           ( ! (let (([[V36:.+]] (= [[V35]] 4)))
+  // CHECK:                               (let (([[V37:.+]] (= [[V34]] 3)))
+  // CHECK:                               (let (([[V38:.+]] (= [[V37]] [[V36]])))
+  // CHECK:                               [[V38]])))
+  // CHECK:                               :pattern ((let (([[V39:.+]] (= [[V34]] 3)))
+  // CHECK:                               [[V39]]) (let (([[V40:.+]] (= [[V35]] 4)))
```

Comment:
It might be a good idea to add `{{$}}` at the end of a line to check that there aren't any extra closing parentheses since we had issues with that before.